### PR TITLE
Add clusterName in Print Column

### PR DESCRIPTION
**Pull request by kashifest**
Adding the clusterName in the print column to depict which cluster the object belongs to. The clusterName is taken from the label.

### CODE REVIEW

**Review by jan-est — COMMENTED** [review]
This looks good to me!

File: api/v1alpha3/baremetalmachine_types.go [hunk 1]
```diff
@@ -141,6 +141,7 @@ type BareMetalMachineStatus struct {
 // +kubebuilder:subresource:status
 // +kubebuilder:printcolumn:name="ProviderID",type="string",JSONPath=".spec.providerID",description="Provider ID"
 // +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="BaremetalMachine is Ready"
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this BMCluster belongs"
```
Comment (maelk) [inline api/v1alpha3/baremetalmachine_types.go]: BMmachine ?

File: config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalmachines.yaml [hunk 2]
```diff
@@ -204,6 +204,10 @@ spec:
       description: BaremetalMachine is Ready
       name: Ready
       type: string
+    - JSONPath: .metadata.labels.cluster\.x-k8s\.io/cluster-name
+      description: Cluster to which this BMCluster belongs
```
Comment (maelk) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalmachines.yaml]: same here, re-generate with the fixed types to fix it.

File: api/v1alpha3/baremetalmachine_types.go (hunk 1 above)
Comment (kashifest) [reply api/v1alpha3/baremetalmachine_types.go]: fixed

File: config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalmachines.yaml (hunk 2 above)
Comment (kashifest) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalmachines.yaml]: fixed

**Review by maelk — APPROVED** [review]

**Pull request merged**